Venv configuration

renjingxiao opened the pull request:
Hi Andrea/Paolo,

I added a myenv folder and a blank requirements.txt. Please correct me if I've made any mistakes.

Regards,
Jean

andreaTP commented:
Hi @renjingxiao and thanks a lot for this PR!

As far as I understand this configuration should be sufficient to:
- commit only `requirements.txt`
- add `.env` to the `.gitignore` file
- remove the `myenv` folder
- add instructions on how to replicate the environment, I expect something along those lines: `virtualenv .env && source .env/bin/activate && pip install -r requirements.txt`

Ideally, but not required, using the virtualenv in CI should guarantee everything is properly automated.

andreaTP commented on README.md:
```diff
@@ -14,6 +14,18 @@ yarn install
+python3.10 -m venv .env
+# activate it
+source .env/bin/activate
+# or this if on windows
```
I would keep the windows instructions in the comment as ppl tend to copy-paste instructions directly in the terminal ...

andreaTP reviewed (commented):
2 minor comments and LGTM, thanks!

renjingxiao commented:
Thank you Andrea! That makes more sense and I have changed them.
I have also moved the environment section before the "Running Locally" section and would like to know if it's better.
Thanks again for your time and patience!

andreaTP reviewed (approved):
Sorry for the delay, LGTM! Thanks a lot!

andreaTP commented:
Hi all and sorry for the late check, I figured out that the current `requirements.txt` is including some non-portable library like: https://github.com/Apicurio/backstage/blob/0e907e279315f97ce766075b1c09119be7fe6c65/requirements.txt#L80

and is probably installing quite a number of dependencies that are not strictly needed.
Please note that I tested in CI on a `macos-latest` runner and locally on my Mac and I cannot reproduce the original issue, unfortunately.
